tools/validate_whitespace.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+import os
+import re
+import sys
+import glob
+import subprocess
+
+
+def load_ignores():
+    ignores = set()
+    fail = False
+    for ind, line in enumerate(open(".wsignore")):
+        line = line.strip()
+        if not line:
+            continue
+        if line.startswith("#"):
+            continue
+        expanded = glob.glob(line, recursive=True)
+        if len(expanded) == 0:
+            print(f".wsignore:{ind + 1}: No matches for line: {line}")
+            fail = True
+        ignores.update(expanded)
+    if fail:
+        sys.exit(1)
+    return ignores
+
+
+def count_trailing_newlines(s):
+    match = re.search(r"(\n+)$", s)
+    return len(match.group(1)) if match else 0
+
+
+def validate_contents(data):
+    if not data:
+        return
+    try:
+        text = data.decode("utf")
+    except Exception as ex:
+        yield f" Failed to decode utf-8: {ex}"
+        return
+
+    count = 0
+    max_trailing_errors = 9
```

Comment:
Why?

Reply:
Some files have too many of these, polluting the output.

Reply:
On second thought, we can live without this and it simplifies the code:L https://github.com/databricks/cli/pull/2603/commits/9cae271d34ebd66b7adb23dc0048865d27c0896c